auxin/src/context.rs

```diff
@@ -174,15 +185,8 @@ custom_error! { pub UnidentifiedAccessError
 
 impl AuxinContext {
 	/// Generate an unidentified-access key for a user who accepts unrestricted unidentified access.
-	///
-	/// # Arguments
-	///
-	/// * `rng` - Mutable reference to a random number generator, which must be cryptographically-strong (i.e. implements the CryptoRng interface).
-	fn get_unidentified_access_unrestricted<R>(&mut self, rng: &mut R) -> crate::Result<Vec<u8>>
-	where
-		R: RngCore + CryptoRng,
-	{
-		let bytes: [u8; 16] = rng.gen();
+	fn get_unidentified_access_unrestricted(&mut self) -> crate::Result<Vec<u8>> {
+		let bytes = [0u8; 16];
```

Comment:
Why remove the random generation?

Reply:
The way group message unidentified access keys work assumes "Unrestricted unidentified access" means all 0s. The group unidentified access key is derived by XORing every member's unidentified access key together, and if somebody's key is random that means it's not going to match what the server has.
See: [The relevant code path in Signal-Server
](https://github.com/signalapp/Signal-Server/blob/master/service/src/main/java/org/whispersystems/textsecuregcm/controllers/MessageController.java#L463)